### PR TITLE
add a delegate to receive rewards

### PR DESCRIPTION
**Pull request by jim4067**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Webb Relayer does not have the ability to to specify another address for rewards. added a way to specify a `reward-address` instead of default account for fee rewards

## What is the new behavior?
Now we are able to specify an address that is different from the one submitting the transaction for fee rewards

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: config/ethereum/ganache.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ http-endpoint = "http://localhost:8545"
 ws-endpoint = "ws://localhost:8545"
 chain-id = 1337
 private-key = "0xc0d375903fd6f6ad3edafc2c5428900c0757ce1da10e5dd864fe387b32b91d7e"
-reward-address = "0x1111111111111111111111111111111111111111111111111111111111111111"
+reward-address = "0x1111111111111111111111111111111111111111"
```
Comment (shekohex) [inline config/ethereum/ganache.toml]:
```suggestion
beneficiary = "0x1111111111111111111111111111111111111111"
```

File: src/config.rs [hunk 2]
```diff
@@ -129,6 +125,9 @@ pub struct CommonContractConfig {
     pub deployed_at: u64,
 }
 
+#[derive(Debug, Clone)]
+pub struct SerializeableAddress(Address);
+
```
Comment (shekohex) [inline src/config.rs:130]:
```suggestion
```

File: src/config.rs [hunk 3]
```diff
@@ -283,7 +282,7 @@ pub fn load<P: AsRef<Path>>(path: P) -> anyhow::Result<WebbRelayerConfig> {
     // and finally deserialize the config and post-process it
     let config = serde_path_to_error::deserialize(cfg);
     match config {
-        Ok(config) => postloading_process(config),
+        Ok(config) => dbg!(postloading_process(config)),
```
Comment (shekohex) [inline src/config.rs]:
```suggestion
        Ok(config) => postloading_process(config),
```

**Review by shekohex — CHANGES_REQUESTED** [review]
Just a few changes and this would be ready to go.

Also, @jim4067 could you also update the docs for the README file?

File: src/config.rs (hunk 3 above)
Comment (jim4067) [reply src/config.rs]: I removed the dbg! macro yesterday.

File: src/config.rs (hunk 2 above)
Comment (jim4067) [reply src/config.rs]: done

**Comment by jim4067** [comment]
no problem @shekohex

File: config/ethereum/rinkeby.toml [hunk 4]
```diff
@@ -4,6 +4,7 @@ ws-endpoint = "wss://rinkeby.infura.io/ws/v3/9aa3d95b3bc440fa88ea12eaa4456161"
 explorer = "https://rinkeby.etherscan.io"
 chain-id = 4
 private-key = "$RINKEBY_PRIVATE_KEY"
+beneficiary = "0x2222222222222222222222222222222222222222"
```
Comment (nepoche) [inline config/ethereum/rinkeby.toml:7]:
```suggestion

```

File: config/harmony/testnet1.toml [hunk 5]
```diff
@@ -27,6 +27,7 @@ chain-id = 1666700001
 # 4. if it doesn't contains special characters and has 12 or 24 words in it
 #    then we should process it as a mnemonic string: 'word two three four ...'
 private-key = "$HARMONYTESTNET1_PRIVATE_KEY"
+#account address for fee rewards (other than the one which is submitting the transaction)
```
Comment (nepoche) [inline config/harmony/testnet1.toml:30]:
```suggestion

```

**Review by nepoche — CHANGES_REQUESTED** [review]

**Review by shekohex — APPROVED** [review]

**Comment by shekohex** [comment]
There is something wrong with the checks, @jim4067 could you try fixing these falling tests?

**Comment by nepoche** [comment]
Need to coordinate a merge / updates with dapp integrations

**Pull request merged**